blocks/bricks/conv.py

```diff
@@ -148,13 +148,26 @@ class ConvolutionalLayer(Sequence, Initializable):
     def __init__(self, filter_size, num_filters, num_channels, pooling_size,
                  activation, conv_step=(1, 1), pooling_step=None,
                  border_mode='valid', **kwargs):
-        convolution = Convolutional(filter_size, num_filters,
-                                    num_channels, conv_step, border_mode)
-        pooling = MaxPooling(pooling_size, pooling_step)
+        self.convolution = Convolutional(filter_size, num_filters,
+                                         num_channels, conv_step, border_mode)
+        self.pooling = MaxPooling(pooling_size, pooling_step)
         super(ConvolutionalLayer, self).__init__(
-            application_methods=[convolution.apply, activation, pooling.apply],
+            application_methods=[self.convolution.apply, activation,
+                                 self.pooling.apply],
             **kwargs)
 
+    @property
+    def output_dim(self):
```

Comment:
In the MLP class `output_dim` is there to set dimension, but the main interface for asking them is `get_dim` (see `Brick` documentation). There is certain philosophy behind all this and I swear to explain it tomorrow! :)